packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -492,6 +473,41 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> {
 		return rebaseResult;
 	}
 
+	/**
+	 * Replace a range of commits on this branch with a single commit composed of equivalent changes.
+	 * @param startCommit - All commits after (but not including) this commit, up to and including the branch's head, will be squashed.
+	 * @remarks The commits after startCommit will be removed from this branch, and the squash commit will become the new head of this branch.
+	 * The change event emitted by this operation will have a `change` property that is undefined, since no net change occurred.
+	 */
+	public squash(startCommit: GraphCommit<TChange>): GraphCommit<TChange>[] {
+		if (startCommit === this.head) {
+			return [];
+		}
+
+		const removedCommits: GraphCommit<TChange>[] = [];
+		const ancestor = findAncestor([this.head, removedCommits], (c) => c === startCommit);
```

Comment:
This is, for now, introducing duplicated work (because "removedCommits" is already stored by the transaction state). However, it is necessary to do this independently if we want to later remove transactions from branches, which is the goal. I don't expect this to actually regress any behavior, except (maybe?) in the case of an absolutely massive transaction.

Reply:
Doesn't bother me :)